lib/jsonapi/utils/underscore.ex

```diff
@@ -0,0 +1,32 @@
+defmodule JSONAPI.Utils.Underscore do
+  @moduledoc """
+  Helpers for replacing underscores with dashes.
+  """
+
+  def underscore?, do: Application.get_env(:jsonapi, :underscore_to_dash, false)
+
+  def underscore(value) when is_atom(value) do
+    value
+    |> to_string
+    |> underscore
+  end
+
+  def underscore(value) when is_binary(value) do
+    value
+    |> String.replace("_", "-")
+  end
+
+  def underscore(value) when is_map(value) do
+    value
+    |> Enum.map(&underscore/1)
+    |> Enum.into(%{})
+  end
+
+  def underscore({key, value}) do
+    if is_map(value) do
```

Comment:
I don't think this check is necessary, do you?  When you pass value through `underscore/1` the previously defined functions should capture the map or the binary.  I don't think there's any reason to _not_ run `value` through `underscore/1`

Reply:
I think it is with the current flow.  Recursively calling underscore on all of the values means it will be called on regular values, e.g.

```
%{
  id: 1,
  my_text: "my_text"
  # ...
}
```

gets serialized to

```
%{
  id: 1,
  my-text: "my-text"
}
```

The two options I see are to either not do it recursively, or do it recursively and avoid calling `underscore` on non-map values.

Reply:
You're right @mitchellhenke, we should leave it as-is 👍